nncf/common/pruning/operations.py

```diff
@@ -244,6 +319,21 @@ class ReshapePruningOp(BasePruningOp):
     def _is_flatten(node: NNCFNode) -> bool:
         return len(node.layer_attributes.output_shape) == 2
 
+    @staticmethod
+    def _is_not_mixing_dim(node: NNCFNode, graph) -> bool:
+        input_shape = node.layer_attributes.input_shape
+
+        output_edges = graph.get_output_edges(node)
+        if not output_edges:
+            return True
```

Comment:
You don't really need this because `layer_attributes` is exist. This is checked in `accept_pruned_input`
```suggestion
```